src/components/EditCoursePage/EditCourseForm.jsx

```diff
@@ -0,0 +1,208 @@
+import React from 'react';
+import { Field, reduxForm } from 'redux-form';
+import PropTypes from 'prop-types';
+
+import RenderInputTextField from '../RenderInputTextField';
+import RenderSelectField from '../RenderSelectField';
+import ImageUpload from '../../components/ImageUpload';
+import RichEditor from '../../components/RichEditor';
+
+
+class EditCourseForm extends React.Component {
+  getOptions() {
+    if (!this.props.courseOptions) {
+      return [];
+    }
+
+    const {
+      data,
+    } = this.props.courseOptions;
+
+    if (!data || !data.actions) {
+      return [];
+    }
+
+    return data.actions.PUT;
+  }
+
+  parseOptions(inChoices) {
+    const outChoices = [];
+    for (let i = 0, len = inChoices.length; i < len; i += 1) {
+      outChoices.push({
+        label: inChoices[i].display_name,
+        value: inChoices[i].value,
+      });
+    }
+
+    return outChoices;
+  }
+
+  render() {
+    const {
+      handleSubmit,
+      number,
+      price,
+      submitting,
+    } = this.props;
+
+    const options = this.getOptions();
+    const levelTypeOptions = options && this.parseOptions(options.level_type.choices);
+    const subjectOptions = options && this.parseOptions(options.subjects.child.choices);
+
+    levelTypeOptions.unshift({ label: '--', value: '' });
+    subjectOptions.unshift({ label: '--', value: '' });
+
+    return (
+      <div className="edit-course-form row">
+        <div className="col">
+          <form onSubmit={handleSubmit}>
+            <Field
+              name="title"
+              component={RenderInputTextField}
+              type="text"
+              label={<strong>Course Title: *</strong>}
+              required
+            />
+            <div className="row">
+              <div id="number" className="col-12">
+                <strong>Number: *</strong>
+              </div>
+              <div className="col-12">{number}</div>
+            </div>
+            <Field
+              name="short_description"
```

Comment:
Should we be using camel case on all of the field names?

Reply:
We use snake case because the name of the field is what is returned when the form is submitted and what we eventually send to discovery. These are written so that they match the discovery fields for updating.